packages/calcite-components/src/components/list-item/list-item.scss

```diff
@@ -147,6 +147,12 @@
   color: var(--calcite-list-icon-color, var(--calcite-color-text-3));
 }
 
+:host([display-mode="flat"][drag-handle][selection-mode="none"]),
```

Comment:
The [`is`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is) function can help simplify this and similar rules:

```scss
:host([display-mode="flat"][drag-handle]:is([selection-mode="none"], [selection-appearance="border"])) {
  .drag-container {
    padding-inline-end: var(--calcite-spacing-xxs);
  }
}
```